### PR TITLE
Add IdentityApi to plugin-search

### PR DESCRIPTION
**Pull request by dariddler**
## Hey, I just made a Pull Request!

Adds IdentityApi to `plugin-search`.

This is needed if you are running an auth-enabled backend for backstage.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/good-jars-turn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-search': minor
```
Comment (benjdlambert) [inline .changeset/good-jars-turn.md:2]: This should be a `patch` release

File: .changeset/good-jars-turn.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+add IdentityApi support
```
Comment (benjdlambert) [inline .changeset/good-jars-turn.md:5]: Maybe something like:

Use the `identityApi` to forward authorization headers to the `search-backend`?

File: plugins/search/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@backstage/plugin-search",
-  "version": "0.4.0",
+  "version": "0.4.1",
```
Comment (benjdlambert) [inline plugins/search/package.json:3]: Seen as we use `changesets` for releases it mean you don't have to worry about updating these version numbers, so you can also rollback this to `HEAD` too 👍

**Comment by dariddler** [comment]
Tests seems to be failing on api-report for `plugin-auth-backend`, which seems to be out of scope of this PR?

**Comment by iamEAP** [comment]
Hmm...  We _just_ merged in a change that adds an `api-report.md` to `@backstage/search-plugin`, and this change would actually affect that.  So maybe make sure your branch has those changes (rebase or merge), and then run `yarn build:api-reports` and you should see an (expected) change to `plugins/search/api-report.md`.  Commit that and push it up and we should be good to go!

Thanks for the contrib.  And the tests!

**Comment by dariddler** [comment]
I pushed the updated api-reports to see if tests start going through, but it feel very strange pushing updated reports for plugins not changed in this PR.

**Comment by benjdlambert** [comment]
@dariddler yep indeed - you have it right, you shouldn't be pushing some of this stuff. We have a PR to address the stuff that you shouldn't be pushing part of this PR here: #6092 

A few good merges made a broken main 😅

**Comment by benjdlambert** [comment]
#6092 is now merged, so you should be able to rebase and it the changes should be much nicer now :)

**Review by benjdlambert — APPROVED** [review]

**Review by iamEAP — APPROVED** [review]
Thanks @dariddler!

**Pull request merged**